src/Vault.sol

```diff
@@ -103,15 +103,19 @@ contract Vault is IVault, VaultToken, Ownable {
 
     /// @inheritdoc IVault
     function take(Currency currency, address to, uint256 amount) external override isLocked {
-        SettlementGuard.accountDelta(msg.sender, currency, amount.toInt128());
+        unchecked {
+            SettlementGuard.accountDelta(msg.sender, currency, amount.toInt128());
+            currency.transfer(to, amount);
```

Comment:
Not sure whether there are any security issues .
Now we update the reservesOfVault after transfer token.
But after we merge the PR 33(reserveOf transient) , should be fine.